platform/entanglement/src/services/LinkService.js

```diff
@@ -45,36 +45,24 @@ define(
             if (parentCandidate.getId() === object.getId()) {
                 return false;
             }
-            if (parentCandidate.getModel().composition.indexOf(object.getId()) !== -1) {
+            if ((parentCandidate.getModel().composition || []).indexOf(object.getId()) !== -1) {
```

Comment:
Our code is littered with extra code for handling invalid inputs, it's overkill.  Passing invalid inputs to a function should blow up clearly.  I recognize there is some need to handle invalid inputs when dealing with scope and digest loops (because control flow due to watches is unpredictable), but code that does not directly interact with scope should not require the same error handling.

If you moved the check for hasCapability('composition') to occur before this if statement, then we don't need to add this fallback for composition being falsy.